From 9e832a6d5072263a1fea554c654e29ccf918e059 Mon Sep 17 00:00:00 2001
From: Nikos Mavrogiannopoulos <nmav@redhat.com>
Date: Thu, 04 Sep 2014 08:18:34 +0000
Subject: corrected regression which caused the failure of octet string extraction

This affected octet strings placed at the end of the structure.
Fixes issue with OCSP response parsing in gnutls.
---
--- a/lib/decoding.c
+++ b/lib/decoding.c
@@ -731,6 +731,7 @@ _asn1_get_octet_string (asn1_node node,
 {
   int len2, len3, counter, tot_len, indefinite;
   int result;
+  int orig_der_len = der_len;
 
   counter = 0;
 
@@ -792,7 +793,7 @@ _asn1_get_octet_string (asn1_node node,
 	  asn1_length_der (tot_len, temp, &len2);
 	  _asn1_set_value (node, temp, len2);
 
-	  ret = _asn1_extract_der_octet (node, der, der_len, flags);
+	  ret = _asn1_extract_der_octet (node, der, orig_der_len, flags);
 	  if (ret != ASN1_SUCCESS)
 	    return ret;
 
